lang/c++/CMakeLists.txt

```
@@ -293,17 +294,28 @@ write_basic_package_version_file(
 configure_package_config_file(
     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/AvroConfig.cmake.in"
     "${CMAKE_CURRENT_BINARY_DIR}/AvroConfig.cmake"
-    INSTALL_DESTINATION lib/cmake/Avro
+    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/Avro
```

```suggestion
    INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/avro-cpp
```
I think we just need to rebase on the latest main branch